### PR TITLE
Add ApiOption overloads to methods on IFollowersClient

### PR DESCRIPTION
**Pull request by SamTheDev**
This implementation targets the issue : #1154 

- [x]  Add an overload for the GetAll* methods on IFollowersClient with the ApiOptions parameter.
- [x]  Implement the the GetAll* method in the FollowersClient class.
- [x]  Add an overload for the GetAll* methods to the IObservableFollowersClient with the ApiOptions parameter.
- [x]  Implement the GetAll* methods in the ObservableFollowersClient class.
- [x]  Implement the needed tests in the FollowersClientTests class.
- [x] Implement the required integration tests in the ObservableFollowersClientTests class.

### CODE REVIEW

File: Octokit.Tests.Integration/Reactive/ObservableFollowersClientTests.cs [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+            {
+                var github = Helper.GetAuthenticatedClient();
+
+                _followersClient = new ObservableFollowersClient(github);
+            }
+
+            [IntegrationTest]       
+            public async Task ReturnsFollowers()
+            {
+                var followers = await _followersClient.GetAllForCurrent().ToList();
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Reactive/ObservableFollowersClientTests.cs]: This one is interesting as my test account doesn't have any followers. Not sure how we should handle this...

Comment (SamTheDev) [inline Octokit.Tests.Integration/Reactive/ObservableFollowersClientTests.cs]: All those passes, i've tried them using a test account with 3 followers!

Comment (shiftkey) [inline Octokit.Tests.Integration/Reactive/ObservableFollowersClientTests.cs]: Oh yes, this isn't your fault. I'm just thinking aloud here about how to make these tests more adaptable to other contributors...

**Comment by SamTheDev** [comment]
Conflicts solved

**Comment by shiftkey** [comment]
@SamTheDev excellent, thanks!

**Pull request merged**